### ISSUE TITLE
Error with MkdirTask

### ISSUE THREAD

**Issue by jawira** [phingofficial/phing#1619]
#### Describe the bug

`MkdirTask` displays errors when creating nested directories in PHP 8.

#### Steps To Reproduce

You can reproduce the problem with the following buildfile:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project name="demo" default="create-dirs">

    <target name="create-dirs">
        <delete dir="foo"/>
        <mkdir dir="foo/bar/baz"/>
    </target>

</project>
```

Execute the target using PHP 8:

```console
$ /usr/bin/php8.0 /usr/local/bin/phing create-dirs
Buildfile: /home/jawira/PhpstormProjects/deleteme/build.xml

demo > create-dirs:

   [delete] Deleting directory /home/jawira/PhpstormProjects/deleteme/foo
[PHP Error] mkdir(): No such file or directory [line 398 of phar:///usr/local/bin/phing/src/Phing/Io/FileSystem.php]
[PHP Error] mkdir(): No such file or directory [line 398 of phar:///usr/local/bin/phing/src/Phing/Io/FileSystem.php]
    [mkdir] Created dir: /home/jawira/PhpstormProjects/deleteme/foo/bar/baz

BUILD FINISHED

Total time: 0.1032 seconds

```

#### Expected behavior

Execute `create-dirs` with PHP 7.4 in order to see the expected behaviour:

```console
$ /usr/bin/php7.4 /usr/local/bin/phing create-dirs
Buildfile: /home/jawira/PhpstormProjects/deleteme/build.xml

demo > create-dirs:

   [delete] Deleting directory /home/jawira/PhpstormProjects/deleteme/foo
    [mkdir] Created dir: /home/jawira/PhpstormProjects/deleteme/foo/bar/baz

BUILD FINISHED

Total time: 0.1020 seconds

```

As you can see in the output, there's no `[PHP Error]`.

#### Additional context

The error occurs in the line:

https://github.com/phingofficial/phing/blob/d98239f270e86c6a886eb93893d5ddd17c9d71bd/src/Phing/Io/FileSystem.php#L398

The problem is the `@ error suppression operator`. In PHP 8 this operator has been modified, some errors are not silenced anymore.

Source: https://php.watch/versions/8.0/fatal-error-suppression

**Comment by siad007** [comment]
@jawira solved 🙇🏼

**Comment by MaximilianKresse** [comment]
The error still exists in the current main branch. 

It looks like the problem is in your error handler. The error handler doesn't handle the silent operator @ correctly: https://github.com/phingofficial/phing/blob/main/src/Phing/Phing.php#L1059


With PHP 8.0 there were a change to the behavior.
[Impact on Error Handlers](https://php.watch/versions/8.0/fatal-error-suppression#suppress-error-handlers) is important for you.

TLDR: Change the (error_reporting() > 0) with something like:
```php
if (!(\error_reporting() & $level)) {
  return true;
}

**Comment by artur-stepien** [comment]
This stil exists in 3.0.1 ... How come it is closed

**Comment by artur-stepien** [comment]
Yeah, but not everyone likes the warnings that are false-positive. Just adding `$recursive` to the `mkdir()` would do the job. Directories created so no reason to suppress the errors in this function.

**Comment by mrook** [comment]
You are misunderstanding me. The directories are already created, so a `recursive` parameter would be superfluous.

And I agree, the warnings are annoying, which is why they will be suppressed.
